plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -919,4 +926,41 @@ public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
         }
         return Optional.empty();
     }
+
+    private boolean schemaExists(SchemaTableName schemaTableName)
+    {
+        return glueClient.getDatabases(new GetDatabasesRequest()).getDatabaseList()
+                .stream().filter(x -> x.getName().equals(schemaTableName.getSchemaName())).findFirst().isPresent();
```

Comment:
nit: we usually put `stream()` on the previous line
and then stream methods on individual lines

```suggestion
        return glueClient.getDatabases(new GetDatabasesRequest()).getDatabaseList().stream()
                .filter(x -> x.getName().equals(schemaTableName.getSchemaName()))
                .findFirst().isPresent();
```

Reply:
Also, my IDE puts a warning here and suggests to simplify it further. Does your display a warning too?

Reply:
Also, we avoid single-letter variable names and other abbreviations (except for `i` and some rare cases). 
Replacing `x` with `database` would be appropriate.

However, I am sure there is a cheaper way to check whether a schema/database exists, without having to list them all

Reply:
Yes, There is another way to check. `getDatabase` will throw EntityNotFoundException exception if databse does not exist. The same way will apply to the table as well. WDYT? And yes IDE displays a warning to simplify the stream.

        try {
            glueClient.getDatabase(new GetDatabaseRequest().withName(schemaTableName.getSchemaName()));
        }
        catch (EntityNotFoundException e) {
            return false;
        }
        return true;